lib/gpsd.js

```diff
@@ -31,30 +31,52 @@ function createServiceSocket(self) {
     sock.on('connect', function (socket) {
         self.logger.info('Socket connected.');
         self.connected = true;
+        self.partialMessages = [];
         self.emit('connected');
     });
 
     sock.on("data", function (payload) {
-        payload = payload.replace(/\}\{/g, '}\n{');
-        var info = payload.split('\n'), data;
-
-        for (var index = 0; index < info.length; index++) {
-            if (info[index]) {
-                if (!self.parse) {
-                    self.emit('raw', info[index]);
-                } else {
-                    try {
-                        data = JSON.parse(info[index]);
-                        self.emit(data.class, data);
-                    } catch (error) {
-                        self.logger.error("Bad message format", info[index], error);
-                        self.emit('error', {
-                            message : "Bad message format",
-                            cause : info[index],
-                            error : error
-                        });
-
-                        continue;
+        if (payload[payload.length - 1] == 10) {
```

Comment:
I suggest changing this line to: `if (payload[payload.length - 1] === '\n') {` this will make it more clear to folks that do not know the ascii table by heart (like me) what character is looked for. ;-) 

```suggestion
        if (payload[payload.length - 1] === '\n') {
```